exec/java-exec/src/main/java/org/apache/drill/exec/physical/impl/join/HashJoinBatch.java

```diff
@@ -560,6 +554,40 @@ public void close() {
     super.close();
   }
 
+  @Override
+  protected void updateBatchMemoryManagerStats() {
+    stats.setLongStat(Metric.LEFT_INPUT_BATCH_COUNT, batchMemoryManager.getNumIncomingBatches(LEFT_INDEX));
```

Comment:
@ppadma - The main motive of moving the metrics inside JoinBatchMemoryManager was to avoid duplicate definition in all the BinaryRecordBatches. I think we should improve on OperatorMetricsRegistry rather than just overriding this method.
Or if you want you can create a new JIRA to track metrics related improvement.

Reply:
Should be fine.